gui/src/app/CompileContext/CompileContextProvider.tsx

```diff
@@ -15,6 +15,34 @@ type CompileContextProviderProps = {
   // none
 };
 
+const useIsConnected = (stanWasmServerUrl: string) => {
+  const probeUrl = `${stanWasmServerUrl}/probe`;
+  const [isConnected, setIsConnected] = useState<boolean>(false);
+  const [retryCode, setRetryCode] = useState<number>(0);
+  const retryConnection = useCallback(() => {
+    setRetryCode((r) => r + 1);
+  }, []);
+  useEffect(() => {
+    setIsConnected(false);
+    if (!probeUrl.startsWith("http://") && !probeUrl.startsWith("https://")) {
+      // important to do this check because otherwise fetch may succeed because
+      // the server of this web app may respond with success
+      return;
+    }
+    (async () => {
+      try {
+        const response = await fetch(probeUrl);
+        if (response.status === 200) {
+          setIsConnected(true);
+        }
+      } catch (err) {
+        setIsConnected(false);
+      }
+    })();
+  }, [probeUrl, retryCode]);
+  return { isConnected, retryConnection };
+};
+
```

Comment:
I don't mean to open up a can of worms by rewriting working code, but it occurs to me that once this is moved into the CompileContext, it may no longer actually need to be a hook. I'm wondering if it would be more natural/clear to write it as just a function or two, with the state (`useState`) held in the CompileContextProvider directly.